app/models/agents/website_agent.rb

```diff
@@ -106,24 +106,41 @@ class WebsiteAgent < Agent
 
       Set `http_success_codes` to an array of status codes (e.g., `[404, 422]`) to treat HTTP response codes beyond 200 as successes.
 
+      The `interpolate` option is the way to format extracted data, and add meta information to each resulted payload.  Its value must be a hash, whose key-value pairs are interpolated after extraction and added to each event payload.  e.g.:
+
+          "interpolate": {
```

Comment:
Hmm, I think it's not very intuitive to create a PostAgent when what you want is scrape a web page.  It should be WebsiteAgent that comes to your mind first.  Also, if you already had a working WebsiteAgent and started to want to add some header value (including the page URL) and/or resolve relative links, you certainly would not want to break up your existing agent just to do those.

I agree what we ideally need would be a combination of something like WebRequestAgent and ParserAgent, but the need for scraping a web page or calling a JSON API is so everywhere and everyday, so we definitely need easy steps to create and edit a series of agents (by extending scenarios?) before considering splitting up, or it would make the lives of ordinary users harder. 😥